File: Semana8/Exercicio4/Aluno.java
package Semana8.Exercicio4;

public class Aluno {
    private String nome;
    private int idade;
    private double[] notas;

    public Aluno(){}

    public Aluno(
        String nome,
        int idade,
        double[] notas
    ){
        this.nome = nome;
        this.idade = idade;
        this.notas = notas;
    }

    public double calcularMedia() {
        double somatorioNotas = 0.0;
        for (double nota : notas) somatorioNotas += nota;
        return somatorioNotas/notas.length;
    }

    public String toString() {
        return "aluno: " + this.nome + "  idade: " + this.idade;
    }
}

File: Semana9/ListaExercicio3/Exercicio4/Aluno.py
from Pessoa import Pessoa

class Aluno(Pessoa):
    def __init__(self, nome: str, notas: list[float]):
        super().__init__(nome)
        self.notas = notas

    def calcular_media(self) -> float:
        if not self.notas:
            return 0.0
        return sum(self.notas) / len(self.notas)

    def __repr__(self) -> str:
        media = self.calcular_media()
        status = "Aprovado" if media >= 7 else "Exame"
        return f"Aluno(nome={self._nome}, notas={self.notas}, media={media:.2f}, status={status})"